x-pack/plugins/file_upload/server/routes.ts

```diff
@@ -37,7 +40,59 @@ function importData(
 /**
  * Routes for the file upload.
  */
-export function fileUploadRoutes(router: IRouter) {
+export function fileUploadRoutes(coreSetup: CoreSetup<StartDeps, unknown>) {
+  const router = coreSetup.http.createRouter();
+
+  router.get(
+    {
+      path: '/api/file_upload/has_import_permission',
+      validate: {
+        query: schema.object({
+          indexName: schema.maybe(schema.string()),
+          checkCreateIndexPattern: schema.maybe(schema.boolean()),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      try {
+        const [, pluginsStart] = await coreSetup.getStartServices();
+        const { indexName, checkCreateIndexPattern } = request.query;
+
+        const authorizationService = pluginsStart.security?.authz;
+        const requiresAuthz = authorizationService?.mode.useRbacForRequest(request) ?? false;
+
+        if (!authorizationService || !requiresAuthz) {
+          return response.ok({ body: { hasImportPermission: true } });
+        }
+
+        const checkPrivilegesPayload: CheckPrivilegesPayload = {
+          elasticsearch: {
+            cluster: ['manage_pipeline'],
+          },
+        };
+        if (checkCreateIndexPattern) {
+          checkPrivilegesPayload.kibana = [
+            authorizationService.actions.savedObject.get('index-pattern', 'create'),
+          ];
+        }
+        if (indexName) {
+          checkPrivilegesPayload.elasticsearch!.index = {
+            [indexName]: ['create', 'create_index'],
+          };
+        }
+
+        const checkPrivileges = authorizationService!.checkPrivilegesDynamicallyWithRequest(
+          request
+        );
+        const checkPrivilegesResp = await checkPrivileges(checkPrivilegesPayload);
+
+        return response.ok({ body: { hasImportPermission: checkPrivilegesResp.hasAllRequested } });
+      } catch (e) {
+        return response.ok({ body: { hasImportPermission: false } });
```

Comment:
I recommend adding a log statement to this catch block. It'll make debugging failures so much easier.